astropy/coordinates/polarization.py

```diff
@@ -0,0 +1,255 @@
+            setattr(out.info, attr, value)
+
+        return out
+
+    def get_sortable_arrays(self):
+        """
+        Return a list of arrays which can be lexically sorted to represent
+        the order of the parent column.
+
+        For StokesCoord this is just the underlying values.
+
+        Returns
+        -------
+        arrays : list of ndarray
+        """
+        return [self._parent._data]
+
+
+class StokesCoord(ShapedLikeNDArray):
+    """
+    A representation of stokes coordinates with helpers for converting to profile names.
+
+    Parameters
+    ----------
+    stokes_values : array-like
+        The numeric values representing stokes coordinates.
+    """
+
+    info = StokesCoordInfo()
+
+    def __init__(self, stokes, copy=False):
+        if isinstance(stokes, type(self)):
+            self._data = stokes._data.copy() if copy else stokes._data
+            self.info = stokes.info
+        else:
+            data = np.asanyarray(stokes)
+
+            if data.dtype.kind == "O":
+                raise ValueError(
+                    "StokesCoord objects can not be initialised with an object array."
+                )
+
+            if data.dtype.kind == "U":
+                self._data = self._from_symbols(data)
+            else:
+                self._data = data.copy() if copy and data is stokes else data
+
+    @property
+    def shape(self):
+        return self._data.shape
+
+    @property
+    def value(self):
+        return self._data
+
+    @property
+    def dtype(self):
+        return self._data.dtype
+
+    def __array__(self, dtype=None):
+        return self._data.astype(dtype, copy=False)
+
+    def _apply(self, method, *args, **kwargs):
+        cls = type(self)
+
+        if callable(method):
+            new = cls(method(self._data, *args, **kwargs))
+        else:
+            new = cls(getattr(self._data, method)(*args, **kwargs))
+
+        # Copy other 'info' attr only if it has actually been defined.
+        # See PR #3898 for further explanation and justification, along
+        # with Quantity.__array_finalize__
+        if "info" in self.__dict__:
+            new.info = self.info
+
+        return new
+
+    @staticmethod
+    def _from_symbols(symbols):
+        """
+        Construct a StokesCoord from strings representing the stokes symbols.
+        """
+        values_array = np.full_like(
+            symbols, UNKNOWN_STOKES_VALUE, dtype=int, subok=False
+        )
+        for stokes_value, symbol in STOKES_VALUE_SYMBOL_MAP.items():
+            values_array[symbols == symbol.symbol] = stokes_value
+
+        if (unknown_values := np.equal(values_array, UNKNOWN_STOKES_VALUE)).any():
+            raise ValueError(
+                f"Unknown stokes symbols present in the input array: {np.unique(symbols[unknown_values])}"
+            )
+
+        return values_array
+
+    @property
+    def symbol(self):
+        """
+        The coordinate represented as strings
+        """
```

Comment:
```suggestion
        """The coordinate represented as strings."""
```

Reply:
eww, is this the accepted formatting for one line docstrings? I hate it :see_no_evil:

Reply:
https://peps.python.org/pep-0257/#one-line-docstrings

Reply:
We can put this off for now, but PEP257 in the form of pydocstyle D200 is a comin' 👹

Reply:
So turning that off on all my repos 🤪

Reply:
I like @nstarman 's suggestion. No point eating up 2 lines just for some triple quotes.